File: common/src/test/java/dev/number6/slack/TestSlackClientComponent.java
package dev.number6.slack;

import dev.number6.slack.dagger.SlackClientModule;
import dev.number6.slack.port.HttpPort;
import dev.number6.slack.port.SlackPort;
import dagger.Component;

import javax.inject.Singleton;

@Component(modules = {
        SlackClientModule.class,
        FakeHttpModule.class,
        FakeAWSSecretsManagerModule.class
})
@Singleton
public interface TestSlackClientComponent {

    SlackPort slackPort();

    HttpPort httpPort();
}

File: common/src/test/kotlin/dev/number6/slack/TestSlackClientComponent.kt
package dev.number6.slack

import dagger.Component
import dev.number6.slack.dagger.SlackClientModule
import dev.number6.slack.port.HttpPort
import dev.number6.slack.port.SlackPort
import javax.inject.Singleton

@Component(modules = [SlackClientModule::class, FakeHttpModule::class, FakeAWSSecretsManagerModule::class])
@Singleton
interface TestSlackClientComponent {
    fun slackPort(): SlackPort
    fun httpPort(): HttpPort
}